### PR TITLE
CLDR-10338 add u locale extension -dx, disable dictionary break for listed scripts

### PR DESCRIPTION
**Pull request by pedberg-icu**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-10338
- [x] Updated PR title and link in previous line to include Issue number

Add u locale extension key -dx to disable dictionary breaks for one or more script codes listed as arguments.
- Add BCP47 info
- Add key name in English
- Update LDML spec
- Update tests; this is a bit incomplete for now.

### CODE REVIEW

File: docs/ldml/tr35.html [hunk 1]
```diff
@@ -2369,6 +2369,29 @@ <h4><a href="#Key_And_Type_Definitions_" name=
...
+        "https://github.com/unicode-org/cldr/tree/latest/common/bcp47/segmentation.xml">segmentation.xml</a>.</strong></td>
+      </tr>
+      <tr class="changed">
+        <td>"dx"</td>
+        <td>Dictionary break script exclusions</td>
+        <td>
+          <i>ISO 15924 script code</i>
+        </td>
+        <td>
+          <p>Items of type SCRIPT_CODE, which are codes consisting of 4 ASCII letters
```
Comment (macchiati) [inline docs/ldml/tr35.html:2390]: Change this to reference a [valid Unicode script subtag](http://unicode.org/reports/tr35/#unicode_script_subtag)

(maybe also other places if you have time)

**Review by macchiati — CHANGES_REQUESTED** [review]
One small item.

Also, should update release page, at least with a TBD

File: docs/ldml/tr35.html (hunk 1 above)
Comment (pedberg-icu) [reply docs/ldml/tr35.html]: Thanks, I did make that change

File: docs/ldml/tr35.html [hunk 2]
```diff
@@ -2369,6 +2369,29 @@ <h4><a href="#Key_And_Type_Definitions_" name=
           <em>Unknown or Invalid Currency</em>.</p>
         </td>
       </tr>
+      <tr class="changed">
+        <td colspan="4"><strong>A <a href=
+        "#UnicodeDictionaryBreakExclusionIdentifier" name=
+        "UnicodeDictionaryBreakExclusionIdentifier" id=
```
Comment (markusicu) [inline docs/ldml/tr35.html:2375]: The anchor names (id=...) in the spec use a variety of styles, but underscores seem common. Do we care?

File: docs/ldml/tr35.html [hunk 3]
```diff
@@ -2369,6 +2369,29 @@ <h4><a href="#Key_And_Type_Definitions_" name=
...
+      <tr class="changed">
+        <td>"dx"</td>
+        <td>Dictionary break script exclusions</td>
+        <td>
+          <i><code><a href="#unicode_script_subtag">unicode_script_subtag</a></code> values</i>
+        </td>
+        <td>
+          <p>One or more items of type SCRIPT_CODE, which are valid
+          <code><a href="#unicode_script_subtag">unicode_script_subtag</a></code> values.</p>
+          <p>The code Zyyy (Common) can be specified to exclude all scripts.</p>
```
Comment (markusicu) [inline docs/ldml/tr35.html:2392]: Add that in that case it needs to be the only code?

File: docs/ldml/tr35.html (hunk 2 above)
Comment (pedberg-icu) [reply docs/ldml/tr35.html:2375]: At least in this table, id values specified as a category are italicized, while explicit values are quoted and non-italic. So this follows that style.

File: docs/ldml/tr35.html (hunk 3 above)
Comment (pedberg-icu) [reply docs/ldml/tr35.html:2392]: Good point, I will update to indicate this.

Comment (pedberg-icu) [reply docs/ldml/tr35.html]: Done...

File: docs/ldml/tr35.html (hunk 2 above)
Comment (markusicu) [reply docs/ldml/tr35.html:2375]: I am talking about the anchor text id="UnicodeDictionaryBreakExclusionIdentifier", not the CSS style of the presentation.
Other anchors use underscores (id="Unicode_Dictionary_Break_Exclusion_Identifier"), lowercase, etc.

Comment (pedberg-icu) [reply docs/ldml/tr35.html:2375]: The camel-case style seems to be used for all of the other id anchors in this table, so I just followed that... e.g.
```
#UnicodeCurrencyFormatIdentifier
#UnicodeCollationIdentifier
#UnicodeCurrencyIdentifier
#UnicodeEmojiPresentationStyleIdentifier
#UnicodeFirstDayIdentifier
...
```

**Comment by pedberg-icu** [comment]
Does this seem OK now? I would like to merge for the CLDR-ICU integration if possible

**Comment by pedberg-icu** [comment]
> Also, should update release page, at least with a TBD

I did add add least  skeleton entry for this in https://sites.google.com/site/cldr/index/downloads/cldr-38

**Review by markusicu — APPROVED** [review]
lgtm tnx

**Pull request merged**